### ISSUE TITLE
MariaDB Support

### ISSUE THREAD

**Issue by ruudvh** [zach-adams/hgv-deploy-full#4]
Have you given any thoughts to switching the current DB (Percona DB) with, or adding support for, MariaDB?

Thank you,

**Comment by dhilditch** [comment]
Is it faster?

On 19 February 2015 at 23:44, Ruud notifications@github.com wrote:

> Have you given any thoughts to switching the current DB (Percona DB) with,
> or adding support for, MariaDB?
> 
> Thank you,
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zach-adams/hgv-deploy-full/issues/4.

## 

_Dave Hilditch_
Founder
www.affiliatewebdesigners.com

Skype: davehilditch
Landline: +44 131 208 1118
Mobile/Cell: +44 7961 552 082
Twitter: @davehilditch https://twitter.com/davehilditch

**Comment by ruudvh** [comment]
Depends how you define faster. Some things are definitely faster, but others are somewhat slower.
MariaDB is supposed to be a MySQL drop-in as well, however, as far as I can read, with the new MariaDB (10) release they have diverted from the MySQL path and thus returning would not be possible - even though this might be 'fixed' in the future.

MariaDB has MyISAM improvements and supports the same InnoDB as Percona DB through Percona's XtraDB. According to the developers MariaDB does everything Percona does and adds additional features. [Here](http://vbtechsupport.com/606/) and [here](http://vbtechsupport.com/657/) are some (very) old charts, these are the only real benchmarks I can find. 

Reason that I am asking is because many other developments are or have switched, I am thinking about adding it myself but did not want to rediscover the wheel incase others have been working on it already.

Thank you,

**Comment by zach-adams** [comment]
Hi @RuudvH, I have looked in MariaDB as a replacement for MySQL, and it looks very promising. The reason I chose to stick with Percona is mostly because the original Mercury project uses it and I wanted to stick as close to their package as possible. 

Here's their answer, and it looks like their actually considering making the switch in the future. If they do I'll be sure to make it to.

https://github.com/wpengine/hgv/issues/42

**Comment by ahansson89** [comment]
You can easily uninstall Percona and replace it with MariaDB

apt-get remove percona-server-server-5.5

Then install MariaDB.

**Comment by xDae** [comment]
hi, I have created a fork with Varnish 4.0 and MariaDB by default, perhaps someone may be interested ;)

**Comment by zach-adams** [comment]
@xDae Very cool, I'll add a link to the readme

**Comment by matthew6688** [comment]
@xDae I have installed your git, when i test the version of Varnish, It shows:
varnishd (varnish-3.0.7 revision f544cd8)

I think it is still Version 3, or am i doing something wrong?

**Comment by xDae** [comment]
@matthew6688 yes, the version is 3, version 4 was inestable when I created the fork, maybe I should to update the repo

**Comment by matthew6688** [comment]
Yes, you can try to updated to version 4.0, if it is not too trouble.

Now with Version 3, there is a problem with wordpress customizor, it keep login out any changes you make in the customizor.

I think there is something wrong with default.vcl, i think you can fix this in the newly update
------------------ Original ------------------
